### DISCUSSION TITLE
User Management connection: Open Connect ID vs older standards

### DISCUSSION THREAD

**Discussion by marcvanandel** [kadaster-labs/sensrnet-home#199] [Pilot Gemeente]
SensRNet Registry Node integrates with an external user management system by the Open Connect ID standard. This is implemented with the [Dex](https://github.com/kadaster-labs/sensrnet-helm-charts#openid-connect) open source container.

Het lijkt erop dat gemeenten nog niet (allemaal) deze (nieuwe?) Open Connect ID standaard beschikbaar hebben in hun user management system (Azure Active Directory meestal). Moeten we hier iets mee doen?

**Comment by kad-busses** [comment]
1 optie is om Dex ander te configureren.

Dex ondersteund meer dan alleen OpenID Connect connecties, zie voor het volledige overzicht: https://dexidp.io/docs/connectors/. Zo is er bijvoorbeeld ondersteuning voor LDAP en SAML 2.0. Wellicht dat deze interfaces al wel beschikbaar zijn bij de gemeentes.

In principe hoeven we zelf niks aan te passen en regelt Dex de tokens verder, maar we hebben er (nog) geen ervaring mee.

**Reply by banaan666** [reply]
Gezien de waarschuwing over de mogelijke deprication van de SAML 2.0 connector binnen Dex vermeld op de pagina
https://dexidp.io/docs/connectors/saml/. Zou ik voorstellen deze niet te gaan ondersteunen.

**Comment by banaan666** [comment]
Even wat achtergrond informatie:
Microsoft ADFS 2016 en ADFS 2018 ondersteunen wel OpenID Connect, maar ze ondersteunen geen van beide CORS headers. Wat vanuit een Webapp natuurlijk wel een vereiste is. Microsoft ADFS 2019 bied wel ondersteuning voor CORS al staat deze standaard uit. Een deel van de functionaliteiten uit ADFS 2019 zijn gebackport naar de oude versies als het gaat om HTTP headers. Helaas hebben ze dat niet met de CORS header functionaliteit gedaan.

https://docs.microsoft.com/en-us/windows-server/identity/ad-fs/operations/customize-http-security-headers-ad-fs

**Comment by marcvanandel** [comment]
Zou [Microsoft OAuth2](https://dexidp.io/docs/connectors/microsoft/) een optie zijn?

**Reply by marcvanandel** [reply]
Waarbij het lijkt dat het inrichten bij Azure naar de portal verschoven is: https://portal.azure.com/#blade/Microsoft_AAD_RegisteredApps/ApplicationsListBlade

**Reply by marcvanandel** [reply]
Hiermee is het in ieder geval voor Gemeente Tilburg gelukt!
